core/src/main/java/io/questdb/griffin/engine/functions/regex/AbstractLikeStrFunctionFactory.java

```diff
@@ -185,6 +238,96 @@ public void toPlan(PlanSink sink) {
         }
     }
 
+    private static class ConstContainsStrFunction extends BooleanFunction implements UnaryFunction {
+        private final boolean caseInsensitive;
+        private final String pattern;
+        private final Function value;
+
+        public ConstContainsStrFunction(Function value, String pattern, boolean caseInsensitive) {
+            this.value = value;
+            this.pattern = pattern;
+            this.caseInsensitive = caseInsensitive;
+        }
+
+        @Override
+        public Function getArg() {
+            return value;
+        }
+
+        @Override
+        public boolean getBool(Record rec) {
+            CharSequence cs = getArg().getStr(rec);
+            if (cs == null) {
+                return false;
+            }
+            if (caseInsensitive) {
```

Comment:
bunch of nits:

considering inheritence is in place, may be we should create different functions for cases of case sensitivity?

Reply:
A never-taken branch is free, so I decided to keep this flag as a field. But I see no problem in split classes: done in 7f27427cda14429f6cab04d22f644e429f685167